packages/cli/commands/generateThemeFieldSelectors.js

```diff
@@ -0,0 +1,210 @@
+          HUBL_EXPRESSIONS
+        );
+        const themeFieldPath = hublStatementsMap[placeholderKey].match(
+          /theme\.[\w|.]*/,
+          'g'
+        );
+        const cssSelectors = cssExpression.match(CSS_SELECTORS_REGEX);
+
+        /**
+         * Try to match a HubL statement to any HubL Variables being used
+         */
+        if (cssSelectors && themeFieldPath) {
+          const cssSelector = cssSelectors[1].replace(/\n/g, ' ');
+          const hublThemePath = themeFieldPath[0];
+
+          if (!themeFieldsSelectorMap[hublThemePath]) {
+            themeFieldsSelectorMap[hublThemePath] = [];
+          }
+
+          if (!themeFieldsSelectorMap[hublThemePath].includes(cssSelector)) {
+            themeFieldsSelectorMap[hublThemePath] = themeFieldsSelectorMap[
+              hublThemePath
+            ].concat(cssSelector);
+          }
+        }
+
+        if (cssSelectors && hublStatement) {
+          const cssSelector = cssSelectors[1].replace(/\n/g, ' ');
+          const hublVariableName = Object.keys(hublVariableMap).find(_hubl => {
+            return hublStatement[0].includes(_hubl);
+          });
+
+          const themeFieldKey = hublVariableMap[hublVariableName];
+
+          /**
+           * If the theme path is referenced directly add selectors
+           */
+          if (themeFieldKey) {
+            if (!themeFieldsSelectorMap[themeFieldKey]) {
+              themeFieldsSelectorMap[themeFieldKey] = [];
+            }
+
+            if (!themeFieldsSelectorMap[themeFieldKey].includes(cssSelector)) {
+              themeFieldsSelectorMap[themeFieldKey] = themeFieldsSelectorMap[
+                themeFieldKey
+              ].concat(cssSelector);
+            }
+          }
+        }
+      });
+
+      return themeFieldsSelectorMap;
+    },
+    {}
+  );
+
+  if (!Object.keys(finalMap).length) {
+    logger.error(
+      'Could not find any selectors associated with any theme fields.'
+    );
+    process.exit(EXIT_CODES.ERROR);
+  }
+  Object.keys(finalMap).forEach(themeFieldKey => {
+    const fieldKey = themeFieldKey.split('.');
+    const selectors = finalMap[themeFieldKey];
+    fieldsJson = setPreviewSelectors(fieldsJson, fieldKey.splice(1), selectors);
+  });
+
+  // Because fields can have nested inheritance we generated inherited selectors
+  // multiple times to make sure all inherted selectors are bubbled up.
+  const maxFieldsDepth = getMaxFieldsDepth();
+  for (let i = 0; i < maxFieldsDepth; i += 1) {
+    fieldsJson = generateInheritedSelectors(fieldsJson, fieldsJson);
+  }
+
+  const selectorsMap = generateSelectorsMap(fieldsJson);
+  const selectorsPath = `${themePath}/selectors.json`;
+
+  fs.writeFileSync(selectorsPath, `${JSON.stringify(selectorsMap, null, 2)}\n`);
+
+  logger.success(
+    `Selectors map generated at ${selectorsPath}. These are best guesses for the selectors for each field and is not exhaustive, please double check before uploading these changes.`
```

Comment:
We probably want to reword this as well haha, same with the errors on line 35 and 175, these were just placeholder text as I was testing the script. Maybe something like:

Line 35: `Unable to find theme's fields.json.`
Line 175: `No selectors found.`
Line 199: `Success! Selectors generated for ${themePath}, please double check the selectors generated at ${selectorsPath} before uploading the theme.`

Also worth localizing these if possible as well.

Reply:
Good call 👀 I'll move these to `en.lyaml`

We can't actually localize these yet since it's a public repo and our localization pipeline only works internally, but we'll eventually ™️  have a way to sync it with an internal repo so it'd be good to keep them there

Reply:
Makes sense, was definitely curious how this would be hooked up to our localization tools haha